i18nを適用

helosshi opened the pull request:
「Rails の i18n を理解する」の課題でi18nを適用し、日本語対応を行いました。

helosshi committed:
i18nを適用

maedana commented on .gitignore:
```diff
@@ -39,3 +39,5 @@
 yarn-debug.log*
 .yarn-integrity
+
+.DS_Store
```
:+1:

maedana commented on .idea/.gitignore:
```diff
@@ -0,0 +1,8 @@
+/dataSources/
+/dataSources.local.xml
+# Editor-based HTTP Client requests
+/httpRequests/
```
`.idea` 配下ってIDEの設定ファイル等ではないですか? 不要だと思うのでこれもコミットしないほうがいいのではないかと思いました。いかがでしょう?

helosshi replied:
.ideaを.gitignoreに追記しました!

maedana replied:
.gitignoreに追加するだけでは既にcommitされてしまったものは残ったままとなります。既にコミット済みのものが、pullrequestの差分に出ないようにしましょうー

maedana commented on app/views/books/edit.html.erb:
```diff
@@ -1,6 +1,6 @@
-<h1>Editing Book</h1>
+<h1><%= t('Editing Book') %></h1>
```
全体的にですが、i18nのキーはスネークケースで書くのが慣習なので直しましょうー。

helosshi replied:
修正しました 🆗

maedana commented on app/views/books/index.html.erb:
```diff
@@ -1,14 +1,14 @@
-      <th>Memo</th>
-      <th>Author</th>
-      <th>Picture</th>
+      <th><%= t('Title') %></th>
```
ActiveRcordのモデル名や属性はこの辺を参照してみましょう。
https://railsguides.jp/i18n.html#active-recordモデルで翻訳を行なう

```ruby
human_attribute_name
```
というメソッドを使って書きたいですね

maedana commented on config/locales/ja.yml:
```diff
@@ -0,0 +1,21 @@
+
+  Books: '本棚'
+  Title: 'タイトル'
+  Memo: 'メモ'
```
上で書きましたが、
```ruby
human_attribute_name
```
を使えば、この辺なくせそうですねー。なぜなら

```
  activerecord:
    attributes:
      book:
        title: 'タイトル'
        memo: 'メモ'
        author: '著者'
        picture: '写真'
```
を参照出来るからです。

helosshi committed:
update i18n #1

helosshi commented on app/views/books/edit.html.erb:
```diff
@@ -1,6 +1,6 @@
-<h1><%= t('Editing Book') %></h1>
+<h1><%= t('activerecord.attributes.book.books')%></h1>
```
@maedana 他の箇所もですが、t('activerecord.attributes.book.hoge')で修正しました。
Book.human_attribute_name(:hoge)を使おうとしたのですが、うまく行かず。。
config/locales/ja.yml にもmodel追記行ったのですが、機能しませんでした。
この場合、どこが不足しているのでしょうか 🙏

maedana replied:
まず、DBのテーブル名(books)や、カラム(titleなど)にあたるものだけの話です。先日も記載しましたが、参考リンクも改めてよく読んでみましょう。

https://railsguides.jp/i18n.html#active-recordモデルで翻訳を行なう

こういったDBのモデル名やカラムと関係ない値を`activerecord.attributes.book.xxx`のようなキーにするのは不適切ですねー。

その上で、`Book.human_attribute_name(:hoge)`と書いたならば
```yml
ja:
  activerecord:
    attributes:
      book:
        hoge: "これが表示されるはず"
```
が適用されるはずですが、どううまく行かないのでしょうか? (違う表示になる、エラーになる、etc...) もう少し詳しい説明なりがないとちょっとアドバイスしにくいです。

helosshi replied:
@maedana
レビューありがとうございます。
以前「機能しない」とコメントした部分ですが、
> t('Book.human_attribute_name(:hoge)')

のように''で囲んでいたのがミスということに気づきました。
> t(Book.human_attribute_name(:hoge))

とすることで反映させることができました。

helosshi committed:
human_attribute_nameの適用(fjord-robocup使用済)

Revert "human_attribute_nameの適用(fjord-robocup使用済)"

This reverts commit 60ecfcd3724d15e0198f92cd3a8df25274134ba6.

human_attribute_nameの適用(fjord-robocup使用済)

Revert "human_attribute_nameの適用(fjord-robocup使用済)"

This reverts commit ebfe53d18ba35f219b809526a1915990fcc52611.

fjord-rubocopによる修正

human_attribute_nameの適用

maedana commented on config/locales/ja.yml:
```diff
@@ -0,0 +1,19 @@
+        author: '著者'
+        picture: '写真'
+        books: '本棚'
+        show: '表示'
```
前回指摘コメント書いてますが、データベースのカラムでないもの(books, show, save...etc)を`activerecord.attributes`配下に書くのは違いますね。カラムではないものは別で書きましょう。なんでもいいんですが、例えば

```yml
views:
  books:
    title: 本棚
    ...
```
とか。

helosshi replied:
反映されなかったところが解決した段階で、満足していました。
構造見直し、反映しました。

maedana commented on test/controllers/books_controller_test.rb:
```diff
@@ -1,44 +1,46 @@
+
 require 'test_helper'
 
 class BooksControllerTest < ActionDispatch::IntegrationTest
```
自動生成の無意味なテストだと思います。ファイル自体不要かと思いますがいかがですか。空っぽというわけではないですが、
https://docs.komagata.org/5676#unnecessary_file

の中身のないテストってものに当たるかと。

helosshi replied:
まさにこれに当たります... 🙇 
削除しました。

maedana commented on test/models/book_test.rb:
```diff
@@ -1,3 +1,5 @@
+
 require 'test_helper'
 
 class BookTest < ActiveSupport::TestCase
```
同上です。現時点ではテストないわけですしファイル自体コミットしないほうがベターかと。いずれテスト書くことがあればファイルを復活させて中身を書けばいいかと思います。

maedana commented on test/system/books_test.rb:
```diff
@@ -1,45 +1,47 @@
-require "application_system_test_case"
+# frozen_string_literal: true
```
同上です。自動生成で機能してないのではないかと思います

helosshi committed:
.ideaディレクトリ,不要なファイル削除

ignoreとかのupdate

helosshi committed:
i18nのattributesとviewsを整理
